### ISSUE TITLE
DiceMetric get_not_nans not used

### ISSUE THREAD

**Issue by rijobro** [Project-MONAI/MONAI#2656]
Setting `get_not_nans=True` for `DiceMetric` doesn't seem to have any affect. Possibly from the recent metric API harmonisation? @Nic-Ma 

```python
dice_metric = DiceMetric(reduction="mean", get_not_nans=True)
dice_metric(y_pred=val_outputs, y=val_labels)
```
example output (only returns the metric, not the NaN count):
```python
tensor([[   nan],
        [   nan],
        [0.6895],
        [   nan],
        [0.9454],
        [0.7776],
        [0.9584],
        [0.9457],
        [0.8841],
        [0.8640]], device='cuda:0')
```

**Comment by Nic-Ma** [comment]
Hi @rijobro ,

Based on the latest API, `aggregate()` API is the real final result, it will return `not_nans`:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/metrics/meandice.py#L97

Thanks.

**Comment by rijobro** [comment]
Ah, I see. thanks!

**Comment by rijobro** [comment]
Ah, I see. thanks!

**Comment by myron** [comment]
I'm also puzzled by changes to DiceMetric, previously it returned dice and non_nans,  currently it only returns dice score (ignoring get_non_nans).  The only way I see to have old behavior is to call aggregate() but it combines results from all previous evaluations (e.g. previous 100 iterations), and I only need to get dice score and non_nans for this current iteration, which seems impossible.

**Comment by Nic-Ma** [comment]
Hi @myron ,

You can call `reset()` to remove previous evaluations, then do evaluation for current iteration and `aggregate()` with `get_not_nans=True`.

Thanks.

**Comment by myron** [comment]
yes, I see that I can do that.  But it looks like a lot of calls, to simply get the current Dice :( 
I suggest we revise dice metric, and also add options to do (sigmoid or softmax before, similar to DiceLoss). I can do it

**Comment by Nic-Ma** [comment]
Hi @myron ,

Because we may also use activation in other cases instead of metrics, we decoupled post-processing and metrics logic. May also consider to decouple the logic in loss functions.
CC @wyli @ericspod .

Thanks.

**Comment by myron** [comment]
ok, closing for now
